server.go

```diff
@@ -445,13 +448,12 @@ func (s *server) peerConnected(conn net.Conn, connReq *connmgr.ConnReq, inbound
 
 	// Now that we've established a connection, create a peer, and
 	// it to the set of currently active peers.
-	p, err := newPeer(conn, connReq, s, peerAddr, inbound)
+	p, err := newPeer(conn, connReq, s, peerAddr, inbound, s.chanDB)
 	if err != nil {
 		srvrLog.Errorf("unable to create peer %v", err)
-		if p.connReq != nil {
-			s.connMgr.Remove(p.connReq.ID())
+		if connReq != nil {
+			s.connMgr.Remove(connReq.ID())
 		}
-		p.Disconnect()
```

Comment:
In this case `p` is nil which cause a panic if error occurred on this stage. Maybe we should always return instance of peer from `newPeer` function, in this case we can return the previous logic.